src/scores/probability/brier_impl.py

```diff
@@ -61,3 +67,155 @@ def brier_score(
+        preserve_dims: Optionally specify which dimensions to preserve when
+            calculating the Fair Brier score. All other dimensions will be reduced. As a
+            special case, 'all' will allow all dimensions to be preserved. In
+            this case, the result will be in the same shape/dimensionality
+            as the forecast, and the errors will be the absolute error at each
+            point (i.e. single-value comparison against observed), and the
+            forecast and observed dimensions must match precisely.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
+            by population, custom).
+        fair_correction: Whether or not to apply the fair Brier score correction. Default is True.
+        threshold_mode: The mode to use when determining threshold exceedance. Default is '>='
+            which means that the threshold is exceeded if the forecast is greater than or equal
+            to the threshold. Alternatively, you can provide ``threshold_mode='>'` which means
+            that the threshold is exceeded if the forecast is strictly greater than the threshold.
+
+
+    Returns:
+        The Brier score for the ensemble forecast.
+
+    Raises:
+        ValueError: if values in ``thresholds`` are not monotonically increasing.
+        ValueError: if ``fcst`` contains the dimension 'threshold'.
+        ValueError: if ``obs`` contains the dimension 'threshold'.
+        ValueError: if ``weights`` contains the dimension 'threshold'.
+        ValueError: if ``ensemble_member_dim`` is 'threshold'.
+        ValueError: if ``ensemble_member_dim`` is not in ``fcst`` dimensions.
+
+    References:
+        - Ferro, C. A. T. (2013). Fair scores for ensemble forecasts. Quarterly
+            Journal of the Royal Meteorological Society, 140(683), 1917–1923.
+            https://doi.org/10.1002/qj.2270
+
+    See Also:
+        - :py:func:`scores.probability.brier_score`
+
+    Examples:
+        Calculate the Brier score for an ensemble forecast for a single threshold:
+
+        >>> import numpy as np
+        >>> import xarray as xr
+        >>> from scores.probability import ensemble_brier_score
+        >>> fcst = xr.DataArray(np.random.rand(10, 10), dims=['time', 'ensemble'])
+        >>> obs = xr.DataArray(np.random.rand(10), dims=['time'])
+        >>> ensemble_brier_score(fcst, obs, ensemble_member_dim='ensemble', thresholds=0.5)
+
+        Calculate the Brier score for an ensemble forecast for multiple thresholds:
+        >>> thresholds = [0.1, 0.5, 0.9]
+        >>> ensemble_brier_score(fcst, obs, ensemble_member_dim='ensemble', thresholds=thresholds)
+
+    """
+    if threshold_mode not in [">=", ">"]:
+        raise ValueError("threshold_mode must be either '>=' or '>'.")
+    if ensemble_member_dim == "threshold":
+        raise ValueError("The ensemble_member_dim is not allowed to be 'threshold'.")
```

Comment:
consider adding an explanation to the ValueError of why it's not allowed to be threshold